mesonbuild/mdist.py

```diff
@@ -109,8 +123,20 @@ def git_clone(src_root, distdir):
         mlog.warning('Repository has uncommitted changes that will not be included in the dist tarball')
     if os.path.exists(distdir):
         shutil.rmtree(distdir)
-    os.makedirs(distdir)
-    subprocess.check_call(['git', 'clone', '--shared', src_root, distdir])
+    repo_root = git_root(src_root)
+    if repo_root.samefile(src_root):
+        os.makedirs(distdir)
+        subprocess.check_call(['git', 'clone', '--shared', src_root, distdir])
+    else:
+        subdir = Path(src_root).relative_to(repo_root)
+        tmp_distdir = distdir + '-tmp'
+        if os.path.exists(tmp_distdir):
+            shutil.rmtree(tmp_distdir)
```

Comment:
probably need the `mesonlib.safe_rmtree` (or whatever it's called)

Reply:
Right, fixed now.